src/map.cpp

```diff
@@ -4729,11 +4729,11 @@ item &map::add_item_at( const tripoint &p,
 item map::water_from(const tripoint &p)
 {
     if( has_flag( "SALT_WATER", p ) ) {
-        item ret( "salt_water", 0 );
+        item ret( "salt_water", 0, std::numeric_limits<long>::max() );
```

Comment:
Well, can you tell why the `qty` parameter of the item constructor is an `int`, when it is used to initialize the `charges` member, which is a `long`?